patroni/utils.py

```diff
@@ -352,3 +352,51 @@ def uri(proto, netloc, path='', user=None):
     path = '/{0}'.format(path) if path and not path.startswith('/') else path
     user = '{0}@'.format(user) if user else ''
     return '{0}://{1}{2}{3}{4}'.format(proto, user, host, port, path)
+
+
+def is_standby_cluster(config):
+    # Check whether or not provided configuration describes a standby cluster
+    return isinstance(config, dict) and (config.get('host') or config.get('port') or config.get('restore_command'))
+
+
+def cluster_as_json(cluster):
+    leader_name = cluster.leader.name if cluster.leader else None
+    xlog_location_cluster = cluster.last_leader_operation or 0
+
+    ret = {'members': []}
+    for m in cluster.members:
+        if m.name == leader_name:
+            config = cluster.config.data if cluster.config and cluster.config.modify_index else {}
+            role = 'standby_leader' if is_standby_cluster(config.get('standby_cluster')) else 'leader'
+        elif m.name == cluster.sync.sync_standby:
+            role = 'sync_standby'
+        else:
+            role = 'replica'
+
+        conn_kwargs = m.conn_kwargs()
+        member = {'name': m.name, 'host': conn_kwargs['host'], 'port': int(conn_kwargs['port']),
+                  'role': role, 'state': m.data.get('state', ''), 'api_url': m.api_url}
+        member.update({n: m.data[n] for n in 'timeline pending_restart scheduled_restart tags'.split() if n in m.data})
```

Comment:
Why `split`, and not having a list right away?

Reply:
Than it won't fit into 120 characters :)

Reply:
Indentation, a separate variable, so many other options.